src/plcc/load_spec/parse_spec/parse_lexical_spec.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import dataclass
+import re
+from re import Match
+
+from ..load_rough_spec.parse_lines import Line
+
+@dataclass
+class LexicalRule:
+    line: Line
+    isSkip: bool
+    name: str
+    pattern: str
+
+@dataclass
+class LexicalSpec:
+    ruleList: [LexicalRule|Line]
+
+def parse_lexical_spec(lines: list[Line]) -> LexicalSpec:
+    return LexicalParser(lines).parseLexicalSpec()
+
+class LexicalParser():
+    def __init__(self, lines: [Line]):
+        self.lines = lines
+        self.patterns = {
+            'skipToken' : re.compile(r'^skip\s+(?P<Name>\S+)\s+(?P<Pattern>((\'\S+\')|(\"\S+\")))\s*(?:#.*)*$'),
+            'tokenToken' : re.compile(r'(?:^token\s+)?(?P<Name>\S+)\s+(?P<Pattern>((\'\S+\')|(\"\S+\")))\s*(?:#.*)*$')
+        }
+        self.spec = LexicalSpec([])
+
+    def parseLexicalSpec(self) -> LexicalSpec:
+        if not self.lines:
+            return self.spec
+        for line in self.lines:
+            if self._isBlankOrComment(line.string.strip()):
+                continue
+            lineIsSkipToken, lineIsRegularToken = self._matchToken(line.string)
+            if lineIsSkipToken or lineIsRegularToken:
+                self.spec.ruleList.append(self._generateTokenRule(line, lineIsSkipToken, lineIsRegularToken))
+            else:
+                self.spec.ruleList.append(line)
+        return self.spec
+
+    def _generateTokenRule(self, line: Line, lineIsSkipToken: Match[str], lineIsRegularToken: Match[str]) -> LexicalRule:
+        if lineIsSkipToken:
+            return self._generateSkipToken(line, lineIsSkipToken['Name'], lineIsSkipToken['Pattern'])
+        elif lineIsRegularToken:
+            return self._generateRegularToken(line, lineIsRegularToken['Name'], lineIsRegularToken['Pattern'])
+
+    def _isBlankOrComment(self, lineStr: str) -> bool:
+        if lineStr == '':
+            return True
+        elif lineStr.startswith('#'):
+            return True
+        else:
+            return False
```

Comment:
This can be reduced to one line to account for all these if else conditions, you can also let this method handle doing the line.string.strip() rather than passing that as the parameter:
```
def _isBlankOrComment(self, line: Line) -> bool:
    return not line.string.strip() or line.string.strip().startswith("#")
```